### ISSUE TITLE
skript-mirror 2.0 development

### ISSUE THREAD

**Issue by btk5h** [btk5h/skript-mirror#135]
This is a tracker issue for skript-mirror 2.0 and will serve as a roadmap for what needs to be done before the first release of skirpt-mirror 2.0.

- [ ] Wait for Skript to become stable. The way Skript's developement is being managed currently does not give me confidence that any API behavior skript-mirror relies on will work the same in the future.
- [ ] Internal refactoring. A lot of the way reflection is done can be cleaned up, and much of the logic for custom elements is due for a rewrite.
- [ ] [Feature implementation.](https://github.com/btk5h/skript-mirror/projects/2)
- [ ] Figure out a funding model. This project will always be open-source and free to use, but that alone is not sustainable.

**Comment by kulkalkul** [comment]
What about Skript independent, standalone version? Would that be complicated to implement?

**Comment by btk5h** [comment]
Many skript-mirror features are dependent on the implementation of Skript features, so that would not be possible.

**Comment by kulkalkul** [comment]
I know, it is out of scope for this addon but, did you ever thought of replicating it, maybe another project, if you had time of course? I don't like Skript's philosophy and think skript-mirror makes Skript bearable to use by patching that philosophy. Skript could provide a way to import java modules and reflection without trying to invent a "language".

**Comment by btk5h** [comment]
I feel like Skript works *because* it rethinks what a programming language is. Skript's high-level abstractions offer an incredibly gentle approach to programming, though that comes at the expense of scalability with larger, more complex projects.

The goal of skript-mirror is to fix a few of those shortcomings by allowing simple Java expressions to be embedded into Skript, but that's not enough to extract into a completely separate language. If you're looking for something that isn't Skript, a more mature language like JavaScript or Lua would probably be your best option.
